src/index.ts

```diff
@@ -200,27 +200,35 @@ class MagicNixCacheAction extends DetSysAction {
       };
     }
 
-    const notifyPort = inputs.getString("startup-notification-port");
-
-    const notifyPromise = new Promise<Promise<void>>((resolveListening) => {
-      const promise = new Promise<void>(async (resolveQuit) => {
-        const notifyServer = http.createServer((req, res) => {
-          if (req.method === "POST" && req.url === "/") {
-            actionsCore.debug(`Notify server shutting down.`);
-            res.writeHead(200, { "Content-Type": "application/json" });
-            res.end("{}");
-            notifyServer.close(() => {
-              resolveQuit();
-            });
-          }
+    const notifyPromise = new Promise<[Promise<void>, string]>(
+      (resolveListening) => {
+        const promise = new Promise<void>(async (resolveQuit) => {
```

Comment:
I know this `async` was here before, but it's not correct and is the source of a great deal of shenanigans:

```suggestion
        const promise = new Promise<void>((resolveQuit) => {
```